rfd/0187/README.md

```diff
@@ -0,0 +1,141 @@
+---
+authors: Nahum Shalman <nahum.shalman@mnx.io>
+state: draft
+---
+
+# RFD 187 AI Contributions to Triton
+
+## Definitions of Terms
+
+AI - Artificial Intelligence. In this document this term will be used to refer to "generative AI" such
+as large language models (LLMs). Commercial examples include ChatGPT from OpenAI and Claude from Anthropic.
+This document intends to cover both use of commercially available tools as well as locally run ones (e.g.
+running a Llama model using ollama).
+
+## Problem statement
+
+If we allow contributions to our repositories that were produced using AI tools, we believe
+that we need to keep track of them. While many of these tools will credit themselves in commit
+messages, our squash-and-merge processes in many of our repos mean we need a plan for how
+to preserve that attribution in the commits that land on our production branches.
+
+## Proposed Solution
+
+### Attribution
+
+- The unit of work that we land is a peer-reviewed GitHub pull request.
+- Only humans count as peer reviewers for approving and merging pull requests.
+- A human proposing a PR with AI generated content is expected to fully understand that content and must review it with extra scrutiny for correctness, security, and adherence to project standards.
+- A PR with ANY AI generated content MUST credit the AI tool(s) used, regardless of the size or significance of the contribution.
+- For repositories where branches are merged preserving the original commits
+  - Commits where the tool credits itself are great. As author or co-author are both acceptable.
+  - If the commit message is being written by a human, make sure to credit the tool appropriately.
+- For repositories where we "squash and merge" down to a single commit
+  - The applied commit message MUST contain an attribution line. We will use `Portions generated by: ` followed by
```

Comment:
Does this mean for single ones the Author: must be a human? I recommend yes (i.e. MUST), and also the "top" commit of a multi-commit merge per above MUST be a human too.

Cf. the picture in https://simonwillison.net/2025/Feb/3/a-computer-can-never-be-held-accountable/

Reply:
I'm not sure what the "top" commit of a multi-commit merge PR refers to.
But let's briefly think through the following case:
A single commit PR is created with that single commit completely authored by an AI tool.
What do we do?
If a person creates a merge commit, they can be the author of the merge commit and responsibility is clear.
The could also squash and merge it and be the author of the squashed commit.
If you look at a75057c51b8515eb153105ad133b7df7dcee5b3e I am still visible as the committer and thus responsible. We could explicitly allow or disallow that.

What should be completely disallowed is a commit with no evidence of which human is taking responsibility for the code change.

I'm still thinking...